pyro/poutine/__init__.py

```diff
@@ -150,16 +157,44 @@ def scale(fn, scale):
     scale factor, scale the score of all sample and observe sites in the
     function.
     """
-    m = ScaleMessenger(scale=scale)
-    # temporary change to preserve API
-    return m(fn) if callable(fn) else m
+    msngr = ScaleMessenger(scale=scale)
+    # XXX temporary compatibility fix
+    return msngr(fn) if callable(fn) else msngr
+
+
+def indep(fn=None, name=None, size=None, dim=None):
+    """
+    Alias for IndepMessenger constructor.
+
+    This messenger keeps track of stack of independence information declared by
+    nested ``irange`` and ``iarange`` contexts. This information is stored in
+    a ``cond_indep_stack`` at each sample/observe site for consumption by
+    ``TraceMessenger``.
+    """
+    msngr = IndepMessenger(name=name, size=size, dim=dim)
+    return msngr(fn) if fn is not None else msngr
+
+
+def enum(fn=None, first_available_dim=None):
```

Comment:
I thought we deleted this since it was not user facing. Is there a reason to expose it?

Reply:
It makes it possible to replace the implementation of `enum` in future without changing the interface